### PR TITLE
New mobile filter on search result 

### PR DESCRIPTION
**Pull request by theomoura**
#### What is the purpose of this pull request? What problem is this solving?

As title says

#### How should this be manually tested?

[Access this workspace](https://newfilter--storecomponents.myvtex.com/shirt)

#### Screenshots or example usage

filtrar

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/FilterNavigator.js [hunk 1]
```diff
@@ -66,6 +73,101 @@ class FilterNavigator extends Component {
...
+    const mapperOption = (isSelected, selectedFilters) => opt => {
+      const pagesArgs = formatFacetToLinkPropsParam(opt.type, opt)
+      return {
+        name: opt.Name,
+        ...pagesArgs,
+        checked: isSelected || !!find(propEq('Name', opt.Name), selectedFilters)
+      }
+    }
+    const mapperOptionsForFilter = (map, rest) => ({ options, type }) => mountOptions(options, type, map, rest).map(mapperOption(false, this.selectedFilters))
+    const reducerForOptions = (current, next) => union(current, next)
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use `flatten` instead of the reducer, it will be cleaner

I think it can substitute the rest operator on line 109 also

https://ramdajs.com/docs/#flatten

File: react/components/SearchResultContainer.js [hunk 2]
```diff
@@ -105,6 +108,9 @@ export default class SearchResultContainer extends Component {
         },
       }
     )
+    // Need to redo this to get the rest and map with the pageProps
```
Comment (luciannojunior) [inline react/components/SearchResultContainer.js]: ??

File: react/components/SearchResultContainer.js [hunk 3]
```diff
@@ -105,6 +108,9 @@ export default class SearchResultContainer extends Component {
...
+    if (!!restParam) pageProps.query.rest = restParam
```
Comment (luciannojunior) [inline react/components/SearchResultContainer.js]: `if (!!restParam) pageProps.query.rest = restParam` can be instead:
`restParam && pageProps.query.rest = restParam`.

Also applies for the next statement

File: react/components/SideBar.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import { IconCaretRight } from 'vtex.styleguide'
+import { injectIntl, intlShape } from 'react-intl'
+import OutsideClickHandler from 'react-outside-click-handler'
+import Animation from 'vtex.store-components/Animation'
+import classNames from 'classnames'
+
+const OPEN_SIDEBAR_CLASS = 'vtex-search-result-sidebar-open'
+
+ /* SideBar component */
+class Sidebar extends Component {
```
Comment (luciannojunior) [inline react/components/SideBar.js]: Isn't this component duplicated of another one in storecomponents or minicart? :thinking: 

If so, just import it instead, preventing code dup

File: react/global.css [hunk 5]
```diff
@@ -185,6 +185,10 @@ body.vtex-filter-popup-open {
   .vtex-search-result__border {
     display: none;
   }
+
+  .vtex-search-result__sidebar {
+    width: 340px;
```
Comment (luciannojunior) [inline react/global.css]: Is this necessary? A percentage or an `auto` wouldn't do the business?

Even so, use `rem` instead

File: react/global.css [hunk 6]
```diff
@@ -212,6 +216,10 @@ body.vtex-filter-popup-open {
   .vtex-gallery--two-columns {
     grid-template-columns: repeat(2, 1fr);
   }
+
+  body.vtex-search-result-sidebar-open {
+    overflow: hidden;
```
Comment (luciannojunior) [inline react/global.css]: Use Tachyons's `overflow-hidden` instead

File: react/package.json [hunk 7]
```diff
@@ -17,9 +17,8 @@
     "react-infinite-scroll-component": "^4.1.0",
     "react-intl": "^2.4.0",
     "react-motion": "^0.5.2",
-    "unorm": "^1.4.1",
-    "react-apollo": "^2.1.3",
-    "react-intl": "^2.4.0"
+    "react-outside-click-handler": "^1.2.2",
```
Comment (luciannojunior) [inline react/package.json]: Why was `react-apollo` and `react-intl` removed?

File: CHANGELOG.md [hunk 8]
```diff
@@ -35,6 +35,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Replace tachyons classes with design tokens.
 - Paddings and margins to match other components.
 - Update loading to use an overlay instead of content loader after initial mount.
+- New layout and behavior of search result filter
```
Comment (lucasecdb) [inline CHANGELOG.md]: Why did you add this here? It should be under the `Unreleased` section

File: react/components/AccordionFilterItem.js [hunk 9]
```diff
@@ -12,57 +13,50 @@ import { getFilterTitle, formatFacetToLinkPropsParam, HEADER_SCROLL_OFFSET } fro
...
+            <Arrow up={open} size={10} />
+          </span>
+        </div>
       </div>
-    </div>
+    }
+    
     {open && (
       <div className="vtex-accordion-filter__item-options">
         {options.map(opt => {
```
Comment (luciannojunior) [inline react/components/AccordionFilterItem.js]: Destructure using `{Name}` since you are only using this prop, and avoid repeating `opt.Name` all over the render

File: react/components/AccordionFilterItem.js [hunk 10]
```diff
@@ -12,57 +13,50 @@ import { getFilterTitle, formatFacetToLinkPropsParam, HEADER_SCROLL_OFFSET } fro
...
-          >
-              {opt.Name}
-
-              {false && (
-                <span className="vtex-accordion-filter__check-icon fr">
-                  <CheckTick />
-                </span>
-              )}
-            </Link>
+            <div className="ph4 pb4 pt5 bb b--muted-4" key={opt.Name}>
```
Comment (luciannojunior) [inline react/components/AccordionFilterItem.js]: Will `Name` always be unique? React will crash with different components with the same key

File: react/components/AccordionFilterContainer.js [hunk 11]
```diff
@@ -48,40 +49,48 @@ class AccordionFilterContainer extends Component {
...
+          const filtersFull = uniqBy(pick(['Name']), [...filtersFlat, ...options])
+          return (
+            <AccordionFilterItem
+              key={filter.title}
+              title={title}
+              options={filtersFull}
+              filtersChecks={filtersChecks}
+              open={isOpen}
+              handleFilterCheck={handleFilterCheck}
+              show={openedItem === null ? true : isOpen}
```
Comment (luciannojunior) [inline react/components/AccordionFilterContainer.js]: `show={!openedItem || isOpen}` would work?

File: react/components/AccordionFilterContainer.js [hunk 12]
```diff
@@ -48,40 +49,48 @@ class AccordionFilterContainer extends Component {
...
-      </Popup>
+        }
+        </div>
+        
+        {nonEmptyFilters.map(filter => {
+          const { type, title, options } = filter
+          const isOpen = openedItem === filter.title
+          
+          const filtersFlat = selectedFilters.filter(({type: selectedType}) => selectedType === type)
+          const filtersFull = uniqBy(pick(['Name']), [...filtersFlat, ...options])
```
Comment (luciannojunior) [inline react/components/AccordionFilterContainer.js]: Full?

File: react/components/AccordionFilterContainer.js [hunk 13]
```diff
@@ -48,40 +49,48 @@ class AccordionFilterContainer extends Component {
...
-                  onOpen={this.handleOpen(filter.title)}
-                  onItemSelected={this.handleItemSelected(onClose)}
-                  oneSelectedCollapse={oneSelectedCollapse}
-                  getLinkProps={getLinkProps}
-                />
-              )
-            })}
+      <div className="vtex-accordion-filter">
+        <div className="pointer flex flex-row items-center pa5 h3 bg-base w-100 z-max bb b--muted-3 bw1">
+          <div className="c-muted-1 pv4 flex items-center" onClick={e => this.setState({ openedItem: null, })}>
```
Comment (luciannojunior) [inline react/components/AccordionFilterContainer.js]: Why receiving the `event` on the onclick?

File: react/components/AccordionFilterContainer.js [hunk 14]
```diff
@@ -1,24 +1,25 @@
...
 import PropTypes from 'prop-types'
 import { injectIntl, intlShape } from 'react-intl'
+import classNames from 'classnames'
+import { uniqBy, pick } from 'ramda'
 
-import Popup from './Popup'
 import AccordionFilterItem from './AccordionFilterItem'
 import { mountOptions } from '../constants/SearchHelpers'
-import FilterIcon from './../images/FilterIcon'
+import Icon from "vtex.use-svg/Icon"
```
Comment (luciannojunior) [inline react/components/AccordionFilterContainer.js]: This import must come before the local ones

File: react/FilterNavigator.js [hunk 15]
```diff
@@ -222,4 +368,4 @@ class FilterNavigator extends Component {
   }
 }
 
-export default withRuntimeContext(FilterNavigator)
+export default withRuntimeContext(injectIntl(FilterNavigator))
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use Ramda's `compose`

File: react/FilterNavigator.js [hunk 16]
```diff
@@ -66,6 +73,101 @@ class FilterNavigator extends Component {
...
+        openContent: !this.state.openContent,
+      })
+    }
+    event.persist()
+  }
+
+  clearFilters = () => {
+    const { getLinkProps, runtime: { navigate } } = this.props
+    const clearedFilters = { ...this.state.filtersChecks }
+    for (const filter in clearedFilters) {
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use `map`instead

File: react/FilterNavigator.js [hunk 17]
```diff
@@ -66,6 +73,101 @@ class FilterNavigator extends Component {
...
+    })
+
+    const checkedFilters = []
+    for (const filter in filterCopy) {
+      if (filterCopy[filter].checked) {
+        checkedFilters.push(filterCopy[filter])
+      }
+    }
+
+    const map = checkedFilters.map(opt => { return getMapByType(opt.type) })
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: No return necessary

File: react/FilterNavigator.js [hunk 18]
```diff
@@ -66,6 +73,101 @@ class FilterNavigator extends Component {
...
+    const { getLinkProps } = this.props
+    const filterCopy = { ...this.state.filtersChecks }
+    filterCopy[filter].checked = !filterCopy[filter].checked
+
+    this.setState({
+      filtersChecks: filterCopy
+    })
+
+    const checkedFilters = []
+    for (const filter in filterCopy) {
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use `filter` and `map` instead

File: CHANGELOG.md (hunk 8 above)
Comment (theomoura) [reply CHANGELOG.md]: You are right, my bad

File: react/components/SearchResultContainer.js (hunk 2 above)
Comment (theomoura) [reply react/components/SearchResultContainer.js]: Just a note for a future improvement

File: react/components/SideBar.js (hunk 4 above)
Comment (theomoura) [reply react/components/SideBar.js]: There is a sidebar on minicart but mine have different structure and design. A future improvement is to create generic sidebar on storecomponents or styleguide to replace all the ones we are using.

File: react/package.json (hunk 7 above)
Comment (theomoura) [reply react/package.json]: There were not, i think they were updated and it is showing like they were removed

File: react/components/AccordionFilterItem.js (hunk 10 above)
Comment (theomoura) [reply react/components/AccordionFilterItem.js]: Yes, always unique for a filter

File: react/components/AccordionFilterContainer.js (hunk 11 above)
Comment (theomoura) [reply react/components/AccordionFilterContainer.js]: Yes

File: react/components/AccordionFilterContainer.js (hunk 12 above)
Comment (theomoura) [reply react/components/AccordionFilterContainer.js]: Bad variable name choices

File: react/components/AccordionFilterContainer.js (hunk 13 above)
Comment (theomoura) [reply react/components/AccordionFilterContainer.js]: removed

File: react/FilterNavigator.js (hunk 16 above)
Comment (theomoura) [reply react/FilterNavigator.js]: Can't do, clearedFilters is an json Object.

File: react/global.css (hunk 6 above)
Comment (lucasecdb) [reply react/global.css]: @luciannojunior better not to, this way we would have an idea of who added the `overflow: hidden` to the body, making debugging easier if we have a problem with this in the future.

File: react/components/SearchResultContainer.js (hunk 3 above)
Comment (theomoura) [reply react/components/SearchResultContainer.js]: Can't do

File: react/FilterNavigator.js [hunk 19]
```diff
@@ -66,6 +72,93 @@ class FilterNavigator extends Component {
...
+  componentDidUpdate(prevProps, prevState) {
+    const nonEmptyFilters = this.mapCheckboxFilters(this.filters.filter(spec => spec.options.length > 0))
+
+    if (this.willUpdateFilters(Object.keys(nonEmptyFilters), Object.keys(prevState.filtersChecks))) {
+      this.setState({ filtersChecks: nonEmptyFilters })
+    }
+  }
+
+  willUpdateFilters = (newFilters, prevFilters) => {
+    if (newFilters.length !== prevFilters.length) return true
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Sort and compare. Check with Ramda

File: react/FilterNavigator.js [hunk 20]
```diff
@@ -66,6 +72,93 @@ class FilterNavigator extends Component {
...
+  handleFilterCheck = (filter) => {
+    const { getLinkProps } = this.props
+    const filterChecksAux = { ...this.state.filtersChecks }
+    filterChecksAux[filter].checked = !filterChecksAux[filter].checked
+
+    this.setState({
+      filtersChecks: filterChecksAux,
+    })
+
+    const checkedFilters = props(Object.keys(filterRamda(x => x.checked, filterChecksAux)), filterChecksAux)
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use `prop` or `path` in the filter

File: react/FilterNavigator.js [hunk 21]
```diff
@@ -66,6 +72,93 @@ class FilterNavigator extends Component {
...
+  state = {
+    openContent: false,
+    filtersChecks: {},
+    pagesArgs: {},
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const nonEmptyFilters = this.mapCheckboxFilters(this.filters.filter(spec => spec.options.length > 0))
+
+    if (this.willUpdateFilters(Object.keys(nonEmptyFilters), Object.keys(prevState.filtersChecks))) {
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use Ramda's `keys` instead of `Object.keys` for conciseness

File: react/FilterNavigator.js [hunk 22]
```diff
@@ -66,6 +72,93 @@ class FilterNavigator extends Component {
...
+    filterChecksAux[filter].checked = !filterChecksAux[filter].checked
+
+    this.setState({
+      filtersChecks: filterChecksAux,
+    })
+
+    const checkedFilters = props(Object.keys(filterRamda(x => x.checked, filterChecksAux)), filterChecksAux)
+    const map = checkedFilters.map(opt => getMapByType(opt.type))
+    map.unshift('ft')
+    const pagesArgs = getLinkProps(checkedFilters, false, checkedFilters.map(opt => { return opt.slug }), map)
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: The return is not necessary.

File: react/FilterNavigator.js [hunk 23]
```diff
@@ -192,15 +255,89 @@ class FilterNavigator extends Component {
...
+              <div className="bottom-0 fl w-50 pl4 pr2">
+                <Button block variation="tertiary" size="default" onClick={() => this.clearFilters()}>{intl.formatMessage({ id: 'search-result.filter-button.clear' })}</Button>
+              </div>
+              <div className="bottom-0 fr w-50 pr4 pl2">
+                <Link
+                  page={pagesArgs.page}
+                  params={pagesArgs.params}
+                  query={pagesArgs.queryString}
+                >
+                  <Button block variation="secondary" size="default">{intl.formatMessage({ id: 'search-result.filter-button.apply' })}</Button>
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use intl's `FormattedMessage` instead

File: react/FilterNavigator.js [hunk 24]
```diff
@@ -192,15 +255,89 @@ class FilterNavigator extends Component {
...
+            <AccordionFilterContainer
+              filters={this.filters}
+              filtersChecks={filtersChecks}
+              onHandleFilterCheck={this.handleFilterCheck}
+              selectedFilters={this.selectedFilters}
+            />
+            <div className="vtex-filter-buttons__box bt b--muted-5 bottom-0 fixed w-100 items-center flex">
+
+              <div className="bottom-0 fl w-50 pl4 pr2">
+                <Button block variation="tertiary" size="default" onClick={() => this.clearFilters()}>{intl.formatMessage({ id: 'search-result.filter-button.clear' })}</Button>
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use intl's FormattedMessage instead

File: react/FilterNavigator.js [hunk 25]
```diff
@@ -192,15 +255,89 @@ class FilterNavigator extends Component {
...
+        <Fragment>
+          <button
+            className={classNames('vtex-filter-popup__button mv0 pointer flex justify-center items-center', {
+              'bb b--muted-1': openContent,
+              'bn': !openContent,
+            })}
+            onClick={event => this.handleClickButton(event)}
+          >
+            <span className="vtex-filter-popup__title c-on-base t-action--small ml-auto">
+              {intl.formatMessage({ id: 'search-result.filter-action.title' })}
```
Comment (luciannojunior) [inline react/FilterNavigator.js]: Use intl's FormattedMessage instead

File: react/components/SearchResultContainer.js [hunk 26]
```diff
@@ -48,8 +48,11 @@ export default class SearchResultContainer extends Component {
     }
   }
 
-  getLinkProps = (spec, useEmptyMapAndRest = false) => {
+  getLinkProps = (spec, useEmptyMapAndRest = false, restParam = null, mapParam = null) => {
     const { rest, map, pagesPath, params } = this.props
+
+
```
Comment (luciannojunior) [inline react/components/SearchResultContainer.js]: Remove all this lines

File: react/components/SideBar.js (hunk 4 above)
Comment (luciannojunior) [reply react/components/SideBar.js]: What's the difference between them? Cant _HOC_ the main `Sidebar` to make the custom behavior?

File: react/FilterNavigator.js [hunk 27]
```diff
@@ -1,15 +1,21 @@
 import React, { Component, Fragment } from 'react'
 import PropTypes from 'prop-types'
-import { flatten, path, identity, contains } from 'ramda'
+import { flatten, path, identity, contains, find, propEq, union, mergeAll, uniqBy, pick, keys, filter as filterRamda, props, map as mapRamda } from 'ramda'
 import ContentLoader from 'react-content-loader'
-import { withRuntimeContext } from 'render'
+import { withRuntimeContext, Link } from 'render'
+import SideBar from './components/SideBar'
+import { Button } from 'vtex.styleguide'
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: absolute imports should come first

File: react/FilterNavigator.js [hunk 28]
```diff
@@ -1,15 +1,21 @@
...
+import FilterIcon from './images/FilterIcon'
+import { FormattedMessage } from 'react-intl'
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: same

File: react/FilterNavigator.js [hunk 29]
```diff
@@ -1,15 +1,21 @@
...
+import classNames from 'classnames'
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: same

File: react/FilterNavigator.js [hunk 30]
```diff
@@ -66,6 +72,94 @@ class FilterNavigator extends Component {
...
+
+    const mapperOption = (isSelected, selectedFilters) => opt => {
+      const pagesArgs = formatFacetToLinkPropsParam(opt.type, opt)
+      return {
+        name: opt.Name,
+        ...pagesArgs,
+        checked: isSelected || !!find(propEq('Name', opt.Name), selectedFilters),
+      }
+    }
+    const mapperOptionsForFilter = (map, rest) => ({ options, type }) => mountOptions(options, type, map, rest).map(mapperOption(false, this.selectedFilters))
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: why do you need this wrapper function since it's only being used once and uses the same value of `map` and `rest` as on the upper scope?

it's generally more readable to place these mapper and filter functions inline, you should only extract them when you need reuse or they're too complex.

File: react/FilterNavigator.js [hunk 31]
```diff
@@ -66,6 +72,94 @@ class FilterNavigator extends Component {
...
+    const reducerForOptions = (current, next) => union(current, next)
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: inline this function

File: react/FilterNavigator.js [hunk 32]
```diff
@@ -66,6 +72,94 @@ class FilterNavigator extends Component {
...
+
+    const filtersAll = [...(filters.map(mapperOptionsForFilter(map, rest)).reduce(reducerForOptions, [])), ...this.selectedFilters.map(mapperOption(true))]
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: this isn't really good to read on a single line.

File: react/components/SearchResultContainer.js [hunk 33]
```diff
@@ -105,6 +105,9 @@ export default class SearchResultContainer extends Component {
         },
       }
     )
+    // Need to redo this to calculate the rest and map in the pageProps
+    restParam && (pageProps.query.rest = restParam)
+    mapParam && (pageProps.query.map = mapParam)
```
Comment (lucasecdb) [inline react/components/SearchResultContainer.js]: not going to pass

**Comment by lucasecdb** [comment]
I'm going to comment more later, but the filter you're applying is wrong.

If you go to any department page, and filter some categories, the first parameter to the `map` query parameter is `ft`, which shouldn't appear at all, since I didn't search for anything.

File: react/FilterNavigator.js [hunk 34]
```diff
@@ -192,15 +256,92 @@ class FilterNavigator extends Component {
...
+              <FormattedMessage id="search-result.filter-action.title" />
+            </span>
+            <span className="vtex-filter-popup__arrow-icon ml-auto pl3 pt2">
+              <FilterIcon size={16} active />
+            </span>
+          </button>
+
+          <SideBar
+            onOutsideClick={this.handleUpdateContentVisibility}
+            isOpen={openContent}>
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: place the `>` in a new line. :pick:

File: react/components/AccordionFilterItem.js [hunk 35]
```diff
@@ -2,67 +2,59 @@ import React, { Fragment } from 'react'
...
   options,
-  type,
   show,
   open,
   onOpen,
-  onItemSelected,
-  getLinkProps,
   intl,
+  filtersChecks,
+  onHandleFilterCheck,
```
Comment (lucasecdb) [inline react/components/AccordionFilterItem.js]: `onFilterCheck`

File: react/components/SearchResultContainer.js (hunk 33 above)
Comment (theomoura) [reply react/components/SearchResultContainer.js]: It is a moment solution. Later we could reformat this part to use the value from pageProps.

File: react/components/AccordionFilterContainer.js [hunk 36]
```diff
@@ -48,40 +48,50 @@ class AccordionFilterContainer extends Component {
...
+          const filtersFlat = selectedFilters.filter(({ type: selectedType }) => selectedType === type)
+          const filtersMerged = uniqBy(pick(['Name']), [...filtersFlat, ...options])
+          return (
+            <AccordionFilterItem
+              key={filter.title}
+              title={title}
+              options={filtersMerged}
+              filtersChecks={filtersChecks}
+              open={isOpen}
+              onHandleFilterCheck={onHandleFilterCheck}
```
[the remark this answers is not in the document]
Comment (theomoura) [reply react/components/AccordionFilterContainer.js]: I changed it name because of a warning from Lint

File: react/components/SearchResultContainer.js (hunk 33 above)
Comment (lucasecdb) [reply react/components/SearchResultContainer.js]: @theomoura but this solution isn't working for all the cases, the only function that should mutate the `map` and `rest` is the `getPagesArgs`

File: react/components/AccordionFilterContainer.js (hunk 36 above)
[the remark this answers is not in the document]
Comment (lucasecdb) [reply react/components/AccordionFilterContainer.js]: @theomoura ignore the lint for this case.

File: react/components/SearchResultContainer.js (hunk 33 above)
Comment (theomoura) [reply react/components/SearchResultContainer.js]: I commited a solution for search and not search cases

**Comment by luciannojunior** [comment]
The functionality works fine, and it was approved by the design. There are some considerations already addressed about the approach for this app that we should change later, that, in some parts, depends of the API.

I will aprove it, but you should pay attention to the issues pointed out by @lucasecdb and try to solve them

File: react/FilterNavigator.js [hunk 37]
```diff
@@ -1,13 +1,13 @@
...
-import { withRuntimeContext, Link } from 'render'
-import SideBar from './components/SideBar'
 import { Button } from 'vtex.styleguide'
-import FilterIcon from './images/FilterIcon'
 import { FormattedMessage } from 'react-intl'
 import classNames from 'classnames'
+import { withRuntimeContext, Link } from 'render'
+import { flatten, path, identity, contains, find, propEq, union, mergeAll, uniqBy, pick, keys, filter as filterRamda, props, map as mapRamda } from 'ramda'
+import SideBar from './components/SideBar'
+import ContentLoader from 'react-content-loader'
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: absolute imports first. also, move the relative ones to the same "block" as the other relative ones.

File: react/FilterNavigator.js [hunk 38]
```diff
@@ -102,10 +102,12 @@ class FilterNavigator extends Component {
         checked: isSelected || !!find(propEq('Name', opt.Name), selectedFilters),
       }
     }
-    const mapperOptionsForFilter = (map, rest) => ({ options, type }) => mountOptions(options, type, map, rest).map(mapperOption(false, this.selectedFilters))
-    const reducerForOptions = (current, next) => union(current, next)
-
-    const filtersAll = [...(filters.map(mapperOptionsForFilter(map, rest)).reduce(reducerForOptions, [])), ...this.selectedFilters.map(mapperOption(true))]
+    
+    const filtersAll = [...(filters.map(({ options, type }) =>
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: indent this properly, please

```js
const filtersAll = [
  ...filters.map(({ options, type }) =>
    mountOptions(options, type, map, rest).map(
      mapperOption(false, this.selectedFilters)
    )
  ).reduce((current, next) => union(current, next), []), 
  ...this.selectedFilters.map(mapperOption(true)),
]
```

File: react/FilterNavigator.js [hunk 39]
```diff
@@ -313,12 +313,13 @@ class FilterNavigator extends Component {
 
           <SideBar
             onOutsideClick={this.handleUpdateContentVisibility}
-            isOpen={openContent}>
+            isOpen={openContent}
+          >
```
Comment (lucasecdb) [inline react/FilterNavigator.js]: remove blank line ⛏

File: react/components/SearchResultContainer.js [hunk 40]
```diff
@@ -107,7 +108,15 @@ export default class SearchResultContainer extends Component {
     )
     // Need to redo this to calculate the rest and map in the pageProps
     restParam && (pageProps.query.rest = restParam)
-    mapParam && (pageProps.query.map = mapParam)
+    if (mapParam) {
```
Comment (lucasecdb) [inline react/components/SearchResultContainer.js]: please, remove these lines, don't hardcode this, it should be handled only by the `getPagesArgs` function

File: react/components/AccordionFilterItem.js [hunk 41]
```diff
@@ -91,6 +93,10 @@ AccordionFilterItem.propTypes = {
   getLinkProps: PropTypes.func,
   /** Intl instance */
   intl: intlShape,
+  /** Filters mapped for checkbox */
+  filtersChecks: PropTypes.object,
```
Comment (lucasecdb) [inline react/components/AccordionFilterItem.js]: remove

File: react/components/FilterSidebar.js [hunk 42]
```diff
@@ -0,0 +1,157 @@
...
+    const {
+      filters,
+      map,
+      rest,
+      getLinkProps,
+    } = this.props
+
+    
+    const mountedFilters = mapRamda(x => 
+      ({ ...x, isSelected: false, link: x.Link }) , selectedFilters)
```
Comment (lucasecdb) [inline react/components/FilterSidebar.js]: move the `link` mapping to the `setState` on line 48 and also inline this map call directly to line 102.

Comment (lucasecdb) [inline react/components/FilterSidebar.js]: rename `x` to something more meaningful

File: manifest.json [hunk 43]
```diff
@@ -29,6 +29,8 @@
     "vtex.styleguide": "8.x",
     "vtex.product-summary": "2.x",
     "vtex.store-graphql": "2.x",
-    "vtex.store": "1.x"
+    "vtex.store": "1.x",
+    "vtex.store-components": "2.x",
```
Comment (brunojdo) [inline manifest.json]: store-components 3.x
